weather_mv/loader_pipeline/bq.py

```diff
@@ -306,6 +338,49 @@ def fetch_geo_point(lat: float, long: float) -> str:
     """Calculates a geography point from an input latitude and longitude."""
     if lat > LATITUDE_RANGE[1] or lat < LATITUDE_RANGE[0]:
         raise ValueError(f"Invalid latitude value '{lat}'")
-    long = ((long + 180) % 360) - 180
     point = geojson.dumps(geojson.Point((long, lat)))
     return point
+
+
+def fetch_geo_polygon(latitude: float, longitude: float, lat_grid_resolution: float, lon_grid_resolution: float) -> str:
+    """Create a Polygon based on latitude, longitude and resolution."""
+    lat_lon_bound = bound_point(latitude, longitude, lat_grid_resolution, lon_grid_resolution)
+    polygon = geojson.dumps(geojson.Polygon([
+        (lat_lon_bound[0][0], lat_lon_bound[0][1]),  # lower_left
+        (lat_lon_bound[1][0], lat_lon_bound[1][1]),  # upper_left
+        (lat_lon_bound[2][0], lat_lon_bound[2][1]),  # upper_right
+        (lat_lon_bound[3][0], lat_lon_bound[3][1]),  # lower_right
+        (lat_lon_bound[0][0], lat_lon_bound[0][1]),  # lower_left
+    ]))
+    return polygon
+
+
+def bound_point(latitude, longitude, lat_grid_resolution, lon_grid_resolution) -> t.List:
+    """Calculate the bound point based on latitude, longitude and grid resolution."""
+    lat_in_bound = latitude in [90.0, -90.0]
+    lon_in_bound = longitude in [-180.0, 180.0]
+
+    lat_range = get_lat_lon_range(latitude, "latitude", lat_in_bound,
+                                  lat_grid_resolution, lon_grid_resolution)
+    lon_range = get_lat_lon_range(longitude, "longitude", lon_in_bound,
+                                  lat_grid_resolution, lon_grid_resolution)
+    lower_left = [lon_range[1], lat_range[1]]
+    upper_left = [lon_range[0], lat_range[1]]
+    upper_right = [lon_range[0], lat_range[0]]
+    lower_right = [lon_range[1], lat_range[0]]
+    return [lower_left, upper_left, upper_right, lower_right]
+
+
+def get_lat_lon_range(value, lat_lon, is_point_out_of_bound, lat_grid_resolution, lon_grid_resolution):
+    """Calculate the latitude, longitude point range point latitude, longitude and grid resolution."""
+    if is_point_out_of_bound:
```

Comment:
Is this necessary if we are also preprocessing the lat/lngs like we do here? 
https://github.com/google/weather-tools/pull/337/files#diff-938d4c3999aa9d0bd9f832240de92dc8ea660bf71f8b9b6d751b3a16773ce4b1R261

Can we merely do the same thing for latitude?

Reply:
Like, given that we have that 1-liner, will the data ever be out of bounds? Can you check?

Reply:
I kindly request a meeting to discuss this logic further and provide clarity on this.

Reply:
I set up a meeting for next week!

Reply:
Example and Logic for the creation of polygon is updated as per your suggestion. Thanks for the input!